apps/web/src/server/applications/case/representations/representation/representation-comment/__tests__/representation-comment.test.js

```diff
@@ -77,5 +77,171 @@ describe('Representation comment page', () => {
+
+				expect(element.innerHTML).toContain('must include a day');
+				expect(dayInput?.outerHTML).toContain('govuk-input--error');
+			});
+
+			it('should show validation error if day and month values are not entered', async () => {
+				const response = await request.post(baseUrl).send({
+					originalRepresentation: 'test',
+					'received-date-day': '',
+					'received-date-month': '',
+					'received-date-year': '2000'
+				});
+
+				const element = parseHtml(response.text);
+				const dayInput = element.querySelector('#received-date-day');
+				const monthInput = element.querySelector('#received-date-month');
+
+				expect(element.innerHTML).toContain('must include a day and month');
+				expect(dayInput?.outerHTML).toContain('govuk-input--error');
+				expect(monthInput?.outerHTML).toContain('govuk-input--error');
+			});
+
+			it('should show validation error if day and year values are not entered', async () => {
+				const response = await request.post(baseUrl).send({
+					originalRepresentation: 'test',
+					'received-date-day': '',
+					'received-date-month': '1',
+					'received-date-year': ''
+				});
+
+				const element = parseHtml(response.text);
+				const dayInput = element.querySelector('#received-date-day');
+				const yearInput = element.querySelector('#received-date-year');
+
+				expect(element.innerHTML).toContain('must include a day and year');
+				expect(dayInput?.outerHTML).toContain('govuk-input--error');
+				expect(yearInput?.outerHTML).toContain('govuk-input--error');
+			});
+
+			it('should show validation error if month value is not entered', async () => {
+				const response = await request.post(baseUrl).send({
+					originalRepresentation: 'test',
+					'received-date-day': '1',
+					'received-date-month': '',
+					'received-date-year': '2000'
+				});
+
+				const element = parseHtml(response.text);
+				const monthInput = element.querySelector('#received-date-month');
+
+				expect(element.innerHTML).toContain('must include a month');
+				expect(monthInput?.outerHTML).toContain('govuk-input--error');
+			});
+
+			it('should show validation error if month and year values are not entered', async () => {
+				const response = await request.post(baseUrl).send({
+					originalRepresentation: 'test',
+					'received-date-day': '1',
+					'received-date-month': '',
+					'received-date-year': ''
+				});
+
+				const element = parseHtml(response.text);
+				const monthInput = element.querySelector('#received-date-month');
+				const yearInput = element.querySelector('#received-date-year');
+
+				expect(element.innerHTML).toContain('must include a month and year');
+				expect(monthInput?.outerHTML).toContain('govuk-input--error');
+				expect(yearInput?.outerHTML).toContain('govuk-input--error');
+			});
+
+			it('should show validation error if year value is not entered', async () => {
+				const response = await request.post(baseUrl).send({
```

Comment:
no test for 'Enter the date' which is what the bug was.

Reply:
typical...